### ISSUE TITLE
Acacia Style Warehouse bugged

### ISSUE THREAD

**Issue by fr3akb0y** [ldtteam/minecolonies#3597]
### Minecolonies version

Version: minecolonies-1.12.2-0.10.280-ALPHA-universal for MC 1.12.2

### Expected behavior

- The Acacia Style Warehouse should work like any other Warehouse.

### Actual behaviour

- The Deliveryman is gettung stuck in the Warehouse (only Acacia style) at the seconds floor of the building (where the building chest is). They wont deposit any stuff and cannot pathfind out of there without breaking blocks (they also get stuck on the roof when removing blocks at the top so he can walk out of the side). That can also apply to builders while building or after finishing it.

![2019-05-08_20 57 08](https://user-images.githubusercontent.com/33865311/57403362-d82a3f00-71d9-11e9-9313-46d29a7b00cb.png)

### Steps to reproduce the problem

1. Build a town with an acacia style Warehouse in it
2. Have at least 1 deliveryman
3. Watch deliveryman getting stuck when he tries to bring resources back to the warehouse
4. ... Profit

**Comment by GanthorPrime** [comment]
the medieval style warehouses are also suffering a similar behavior: ocasionaly, delivery man goes to the terrace at the second floor and try to go down by the roof instead of using the stairs.  he stuck for a while before walking of the roof and taking some damage. this goes to the point the dman dies from falling at that location.

This problems seems to be introduced whith a feature thats makes dman move around the warehouse to simulate picking / storing items in a more realistic way.

maybe a partial solution can be to put some waypoints inside the warehouse but not sure if he will follow waypoints when simulating picking

i will order the builder to build some waypoints (i thing there is a waypoint in the infraestructure decorations folder) and return here to tell

**Comment by fr3akb0y** [comment]
I've now gone ahead and tried that with the waypoints but unfortunately it didnt work. He just walked between the warehouse block and the nearest waypoint. Also I noticed that the thing with not depositing their stuff is not only in the acacia warehouse. I startet a new world with the Medieval Oak style and the deliverymans there are not returning their stuff either. They also get up to 2-5 stacks of items at level 1 and I dont think they changed the regular 1 stack at level 1.
Edit: And I just noticed that when the dman delivers resources for a request he not only delivers the needed resources, he also dumps everything he has in the target chest.
Edit2: after further investigation and upgrading to the latest release version i noticed that the dman not deploying resources might be because of the Racks. I noticed before that in the newest alpha versions they are not connecting to each other. Even upgrading to the release version didnt connect them. Only after rebuilding they did. While they werent connected the dman also threw all his stuff into the dman chest and not in the warehouse.
